bounded/bounded.cabal

```diff
@@ -174,7 +174,6 @@ test-suite test
     , mtl
     , scientific
     , should-not-typecheck
-    , source-constraints
```

Comment:
@epicallan I do not see a common code change assocaited with this, commit scheduling error?

Reply:
So `common/package.yaml` already conditionally adds `source-constraints` as a dependence. This commit removes the duplicate addition of `source-constraints` via individual `packages.yaml` configs. 

Maybe I should reword the commit to: `Remove duplicate source-constraints`

Reply:
yes but the change to `common/package.yaml` is not on that commit?

Reply:
`common/package.yaml` already had `source-constraints` as a conditional dependency based on the development flag so there was no need to change it before this change. 

The other change I made to `common/package.yaml` was to make `source-constraints`  dependent on the `GHC` version. I don't believe that change needs to be part of this commit since it's a new change altogether.